backend/infrahub/git/integrator.py

```diff
@@ -414,7 +413,7 @@ async def get_repository_config(self, branch_name: str, commit: str) -> Optional
         branch_wt = self.get_worktree(identifier=commit or branch_name)
 
         config_file_name = ".infrahub.yml"
-        config_file = Path(os.path.join(branch_wt.directory, config_file_name))
+        config_file = Path(branch_wt.directory, config_file_name)
```

Comment:
`branch_wt.directory` should already be a `Path` object so `config_file = branch_wt.directory / config_file_name` should be enough.